java/vector/src/test/java/org/apache/arrow/vector/TestValueVector.java

```diff
@@ -2167,140 +2279,179 @@ public void testVectorLoadUnload() {
-      assertArrayEquals(STR6, vector.get(5));
-      assertEquals(0, vector.getValueLength(6));
-      assertEquals(0, vector.getValueLength(7));
-      assertEquals(0, vector.getValueLength(8));
-      assertEquals(0, vector.getValueLength(9));
-      assertArrayEquals(STR1, vector.get(10));
-      assertArrayEquals(STR2, vector.get(11));
-      assertEquals(0, vector.getValueLength(12));
-      assertEquals(0, vector.getValueLength(13));
-      assertEquals(0, vector.getValueLength(14));
-
-      /* Check offsets */
+  private void testFillEmptiesUsageHelper(AbstractVariableWidthVector vector) {
+    vector.allocateNew(1024 * 10, 1024);
+
+    setBytes(0, STR1, vector);
+    setBytes(1, STR2, vector);
+    setBytes(2, STR3, vector);
+    setBytes(3, STR4, vector);
+    setBytes(4, STR5, vector);
+    setBytes(5, STR6, vector);
+
+    /* Check current lastSet */
+    assertEquals(-1, vector.getLastSet());
+
+    /* Check the vector output */
+    assertArrayEquals(STR1, vector.get(0));
+    assertArrayEquals(STR2, vector.get(1));
+    assertArrayEquals(STR3, vector.get(2));
+    assertArrayEquals(STR4, vector.get(3));
+    assertArrayEquals(STR5, vector.get(4));
+    assertArrayEquals(STR6, vector.get(5));
+
+    vector.setLastSet(5);
+    /* fill empty byte arrays from index [6, 9] */
+    vector.fillEmpties(10);
+
+    /* Check current lastSet */
+    assertEquals(9, vector.getLastSet());
+
+    /* Check the vector output */
+    assertArrayEquals(STR1, vector.get(0));
+    assertArrayEquals(STR2, vector.get(1));
+    assertArrayEquals(STR3, vector.get(2));
+    assertArrayEquals(STR4, vector.get(3));
+    assertArrayEquals(STR5, vector.get(4));
+    assertArrayEquals(STR6, vector.get(5));
+    assertEquals(0, vector.getValueLength(6));
+    assertEquals(0, vector.getValueLength(7));
+    assertEquals(0, vector.getValueLength(8));
+    assertEquals(0, vector.getValueLength(9));
+
+    setBytes(10, STR1, vector);
+    setBytes(11, STR2, vector);
+
+    vector.setLastSet(11);
+    /* fill empty byte arrays from index [12, 14] */
+    vector.setValueCount(15);
+
+    /* Check current lastSet */
+    assertEquals(14, vector.getLastSet());
+
+    /* Check the vector output */
+    assertArrayEquals(STR1, vector.get(0));
+    assertArrayEquals(STR2, vector.get(1));
+    assertArrayEquals(STR3, vector.get(2));
+    assertArrayEquals(STR4, vector.get(3));
+    assertArrayEquals(STR5, vector.get(4));
+    assertArrayEquals(STR6, vector.get(5));
+    assertEquals(0, vector.getValueLength(6));
+    assertEquals(0, vector.getValueLength(7));
+    assertEquals(0, vector.getValueLength(8));
+    assertEquals(0, vector.getValueLength(9));
+    assertArrayEquals(STR1, vector.get(10));
+    assertArrayEquals(STR2, vector.get(11));
+    assertEquals(0, vector.getValueLength(12));
+    assertEquals(0, vector.getValueLength(13));
+    assertEquals(0, vector.getValueLength(14));
+
+    /* Check offsets */
+    if (vector instanceof BaseVariableWidthVector) {
```

Comment:
same as above, is this okay or keep separate methods for each vector type with certain degree of duplication?

Reply:
This has been resolved.